components/tabs/tabset.component.ts

```diff
@@ -84,12 +87,41 @@ export class Tabset implements OnInit {
       return;
     }
     // Select a new tab if the tab to be removed is selected and not destroyed
-    if (tab.active && this.tabs.length > 1) {
-      // If this is the last tab, select the previous tab. else, the next tab.
-      let newActiveIndex = index === this.tabs.length - 1 ? index - 1 : index + 1;
+    if (tab.active && this.hasAvailableTabs(index)) {
+      let newActiveIndex = this.getClosestTabIndex(index);
       this.tabs[newActiveIndex].active = true;
     }
 
-    this.tabs.slice(index, 1);
+    tab.removed.emit(tab);
+    this.tabs.splice(index, 1);
+  }
+
+  private getClosestTabIndex (index:number):number {
+    let tabsLength = this.tabs.length;
+    if (tabsLength) {
```

Comment:
Yes, but that case still exists in the same place
Check please, source of function [removeTab](https://github.com/vvpanchenko/ng2-bootstrap/blob/master/components/tabs/tabset.component.ts#L84)